epsagon/runners/django.py

```diff
@@ -0,0 +1,69 @@
+"""
+Runner for a Django Python function
+"""
+
+from __future__ import absolute_import
+import uuid
+from ..event import BaseEvent
+from ..utils import add_data_if_needed
+
+
+class DjangoRunner(BaseEvent):
+    """
+    Represents Python Django event runner.
+    """
+
+    ORIGIN = 'runner'
+    RESOURCE_TYPE = 'python_django'
+    OPERATION = 'request'
+
+    def __init__(self, start_time, request):
+        """
+        Initialize.
+        :param start_time: event's start time (epoch).
+        :param request: the incoming request.
+        """
+        super(DjangoRunner, self).__init__(start_time)
+
+        self.event_id = str(uuid.uuid4())
+        self.resource['name'] = request.get_full_path()
+        self.resource['operation'] = request.method
+
+        self.resource['metadata'] = {
+            'Path': request.path,
+            'User Agent': request.headers.get('User-Agent', 'N/A'),
+        }
+
+        if request.body:
+            add_data_if_needed(
+                self.resource['metadata'],
+                'Request Data',
+                request.body
+            )
+
+        add_data_if_needed(
+            self.resource['metadata'],
+            'Request Headers',
+            dict(request.headers)
+        )
+
+    def update_response(self, response):
+        """
+        Adds response data to event.
+        :param response: WSGI Response
+        :return: None
+        """
+        add_data_if_needed(
+            self.resource['metadata'],
+            'Response Data',
+            response.content
+        )
+
+        add_data_if_needed(
+            self.resource['metadata'],
+            'Response Headers',
+            dict(response.items())
+        )
+
+        if response.status_code >= 300:
```

Comment:
It should be 500 - fix also for flask.
(This is also what we do in node)

Reply:
K

Reply:
Why 500 instead of 400?
I get why not 300, cause 300 are redirect codes